addons/misc/functions/fnc_suitcaseNukeZeus.sqf

```diff
@@ -0,0 +1,211 @@
+    _device setVariable [QGVAR(suitcaseNuke_codeTarget), _target];
+    _device addMPEventHandler ["MPKilled", {
+        params ["_device"];
+        if(isServer) then {
+            switch(_device getVariable QGVAR(suitcaseNuke_activate)) do {
+                case "explosive_small": {
+                    private _explosive = createVehicle ["DemoCharge_Remote_Ammo_Scripted", getPosATL _device, [], 0, "CAN_COLLIDE"];
+                    _explosive setDamage 1;
+                    deleteVehicle _device;
+                };
+                case "explosive_large": {
+                    private _explosive = createVehicle ["Bo_GBU12_LGB", getPosATL _device, [], 0, "CAN_COLLIDE"];
+                    _explosive setDamage 1;
+                    deleteVehicle _device;
+                };
+                case "explosive_nuke": {
+                    ["zen_modules_moduleNuke", [ASLToAGL(getPosASL _unit), 200, 300, false]] call CBA_fnc_globalEvent;
+                    deleteVehicle _device;
+                };
+                case "smoke_yellow": {
+                    private _smoke = createVehicle ["SmokeShellYellow_Infinite", getPosATL _device, [], 0, "CAN_COLLIDE"];
+                    hideObjectGlobal _smoke;
+                    _device setVariable [QGVAR(suitcaseNuke_smoke), _smoke];
+                    _smoke attachTo [_device];
+                    _device addEventHandler ["Deleted", {
+                        params ["_device"];
+                        deleteVehicle (_device getVariable [QGVAR(suitcaseNuke_smoke), objNull]);
+                    }];
+                };
+                default {};
+            };
+        };
+
+        // private _customCode = _device getVariable [QGVAR(suitcaseNuke_code), ""];
+        // if(_customCode isNotEqualTo "") then {
+        //     private _codeTarget = _unit getVariable [QGVAR(suitcaseNuke_codeTarget), 0];
+
+        //     if((_codeTarget == 0 and hasInterface) || {
+        //         (_codeTarget == 1 and isServer) || {
+        //         (_codeTarget == 2)}}
+        //     ) then {
+        //         [_device] call (compile _customCode);
+        //     };
+        // };
+    }];
+
+
+    // Defusal Code
+    if(_defuser > 0) then {
+
+        // TODO: with ace, also (optionally) require DefusalKit item
+        private _ace = (isClass(configFile >> "CfgPatches" >> "ace_main"));
+        private _condition = switch(true) do {
+            case (_defuser == 1 and _ace) : { " and ([player] call ace_common_fnc_isEOD)" }; // EOD
+            case (_defuser == 1 and !_ace) : { " and (player getUnitTrait ""ExplosiveSpecialist"")" }; // EOD
+            case (_defuser == 2 and _ace) : { " and (([player] call ace_common_fnc_isEngineer) or ([player] call ace_common_fnc_isEOD)" }; // Engineer
+            case (_defuser == 2 and !_ace) : { " and ((player getUnitTrait ""Engineer"") or (player getUnitTrait ""ExplosiveSpecialist""))" }; // Engineer
```

Comment:
if writing code in text use the macro ``QUOTE()`` instead. Will put whatever is inside ``()`` in quotes on pre-process.